index.js

```diff
@@ -164,37 +164,38 @@ exports.float64 = {
 exports.buffer = {
   preencode (state, b) {
     if (b) {
-      uint.preencode(state, b.length)
-      state.end += b.length
+      uint.preencode(state, b.byteLength)
+      state.end += b.byteLength
     } else {
       state.end++
     }
   },
   encode (state, b) {
     if (b) {
-      uint.encode(state, b.length)
+      uint.encode(state, b.byteLength)
       state.buffer.set(b, state.start)
-      state.start += b.length
+      state.start += b.byteLength
     } else {
       state.buffer[state.start++] = 0
     }
   },
   decode (state) {
     const len = uint.decode(state)
     if (len === 0) return null
-    const b = state.buffer.subarray(state.start, state.start += len)
-    if (b.length !== len) throw new Error('Out of bounds')
+    if (state.end - state.start < len) throw new Error('Out of bounds')
+    const b = state.buffer.subarray(state.start, state.start + len)
```

Comment:
`return state.buffer.subarray(state.start, state.start += len)` ? should be safe with your check above

Reply:
Standard doesn't like returning an assignment expression 😕

Reply:
ah i think that's fine if you wrap it in a paren

Reply:
Giving it a shot 👍

Reply:
Still no dice 😔

Reply:
Bummer, we can keep it as is then.